packages/client/src/runtime/object-enums.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * List of Prisma enums that must use unique objects instead of strings as their values.
+ */
+export const objectEnumNames = ['JsonNullValueInput', 'NullableJsonNullValueInput', 'JsonNullValueFilter']
+
+/**
+ * Module-private symbol used to distinguish between instances of
+ * `ObjectEnumValue` created inside and outside this module.
+ */
+const secret = Symbol()
+
+/**
+ * Base class for unique values of object-valued enums.
+ */
+export abstract class ObjectEnumValue {
+  #representation: string
+
+  constructor(arg?: symbol) {
```

Comment:
What are the benefits of using a secret vs a boolean? I have the feeling have I missed something here...

Reply:
With a boolean it would still be possible to ignore the types and do `new Prisma.NullTypes.JsonNull(true)` in the application code, with a symbol we can't do that even in Prisma code outside this module